mam/v2/mam/message.c

```diff
@@ -644,7 +640,12 @@ void mam_msg_send_packet(mam_msg_send_context_t *ctx,
                                      checksum, ctx->mss, trits_size(payload))));
 
   /*  absorb long trint ord; */
-  pb3_wrap_absorb_longtrint(&ctx->spongos, b, ctx->ord);
+  {
```

Comment:
I see the following alternatives here:
- `pb3_wrap_absorb_longtrint(&ctx->spongos, &tag, ctx->ord);`
- `pb3_wrap_absorb_longtrint(&ctx->spongos, &ord, ctx->ord);`
- `pb3_absorb_external_longtrint(&ctx->spongos, ctx->ord);`
This way encodings and absorptions are localized in PB3 layer. I would use the first options as it's the only place that encodes `ord` as a part of `tag`, otherwise you need to keep in sync absorption of `ctx->ord` here and encoding of `ctx->ord` into `tag` in some other place.